### PR TITLE
optimized tips when sending large size message and bug fix

### PR DESCRIPTION
**Pull request by dugenkui03**
**Make sure set the target branch to `develop`**

## What is the purpose of the change
Details in #6605

### CODE REVIEW

File: client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java [hunk 1]
```diff
@@ -534,6 +535,13 @@ public void executeAsyncMessageSend(Runnable runnable, final Message msg, final
         boolean isSemaphoreAsyncNumAquired = false;
         boolean isSemaphoreAsyncSizeAquired = false;
         int msgLen = msg.getBody() == null ? 1 : msg.getBody().length;
+        if (msgLen > this.getDefaultMQProducer().getBackPressureForAsyncSendSize()) {
```
Comment (superhx) [inline client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java:538]: image

Inflight messages sum size is large than backPressureForAsyncSendSize doesn't mean MESSAGE_ILLEGAL.

MESSAGE_ILLEGAL is usually used when serialized Message size large than maxMessageSize or serialized Message properties size large than 65536 or other Message property conflict condition.

Comment (Focus-rth) [reply client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java:538]: I think backPressureForAsyncSendSize is the total size of buffer message,  not the size of one message

Comment (dugenkui03) [reply client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java:538]: @superhx   fail, if msgLen is large than backPressureForAsyncSendSize.  Is there any suggession other than `backPressureForAsyncSendSize`?
image

@Focus-rth yes.

Comment (superhx) [reply client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java:538]: I think origin checks is correct:
- when inflight messages buffer is larger than backPressureForAsyncSendSize, it suggests client send too fast than broker can take , client should fast return RemotingTooMuchRequestException.
- when single message serialized size is larger than maxMessageSize, client should return MESSAGE_ILLEGAL.

Comment (dugenkui03) [reply client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java:538]: 单个消息大于`backPressureForAsyncSendSize`时、消息永远无法发送成功，这种情况是否使用 `MESSAGE_ILLEGAL` 标识消息不合法更准确呢？

Comment (dugenkui03) [reply client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java:538]: 原来的消息`send message tryAcquire semaphoreAsyncSize timeout`并没有说明获取` tryAcquire semaphoreAsyncSize`的确切原因、重试是否有可能成功 或 总是失败，失败时重试则毫无意义。

PR异常消息 1. 标识消息不合法所以发送终是失败 2. 给出了 最大消息阈值（backPressureForAsyncSendSize一定程度上承担了 maxMessageSize 的作用

Comment (superhx) [reply client/src/main/java/org/apache/rocketmq/client/impl/producer/DefaultMQProducerImpl.java:538]: 那合理的改法应该是
1. 前面会有 maxMessageSize 的判断，返回 MESSAGE_ILLEGAL
2. 然后额外校验 backPressureForAsyncSendSize 和 maxMessageSize 的设置大小关系，避免 backPressureForAsyncSendSize < maxMessageSize

**Pull request closed without merging**